File: awa/sdk_dist/csharp/Awa.Client/Utilities/Workflow/ExecuteAgentWorkflow.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;
using Awa.Client.Models;
using Temporalio.Workflows;

namespace Awa.Client.Utilities.Workflow
{
    public static partial class WorkflowUtilities
    {
        /// <summary>
        /// Execute an agent with the specified configuration.
        /// </summary>
        /// <param name="agentConfig">The configuration for the agent to execute.</param>
        /// <param name="timeoutSeconds">Optional timeout in seconds for agent execution. If not provided, uses agent_config.timeout_seconds or default.</param>
        /// <param name="name">Optional name for the agent execution, used in workflow ID generation.</param>
        /// <returns>The result of the agent execution.</returns>
        public static async Task<TaskResponseModel?> ExecuteAgentWorkflow(
            AgentConfiguration agentConfig,
            int? timeoutSeconds = null,
            string? name = null)
        {
            if (!agentConfig.TimeoutSeconds.HasValue)
            {
                agentConfig.TimeoutSeconds = timeoutSeconds ?? AwaConstants.AgentTimeoutSeconds;
            }

            var result = await Temporalio.Workflows.Workflow.ExecuteChildWorkflowAsync<TaskResponseModel?>(
                AwaConstants.WorkflowExecuteAgent,
                new object?[] { agentConfig },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    Id = name != null ? $"{name}-{Temporalio.Workflows.Workflow.Info.WorkflowId}" : null
                });

            return result;
        }
    }
}

File: awa/awa/sdk/utils/workflow/execute_agent_workflow.py
from temporalio import workflow

from awa.sdk import constants
from awa.sdk.models.agent_modes.agent_configuration import AgentConfiguration
from awa.sdk.models.agent_modes.task_response_model import TaskResponseModel


async def execute_agent_workflow(
    agent_config: AgentConfiguration,
    timeout_seconds: int | None = None,
    name: str | None = None,
) -> TaskResponseModel:
    """Execute an agent with the specified configuration.

    Args:
        agent_config: The configuration for the agent to execute.
        timeout_seconds: Optional timeout in seconds for agent execution.
            If not provided, uses agent_config.timeout_seconds or default.
        name: Optional name for the agent execution, used in workflow ID generation.

    Returns:
        TaskResponseModel: The result of the agent execution.

    """
    if not agent_config.timeout_seconds:
        agent_config.timeout_seconds = timeout_seconds or constants.AGENT_TIMEOUT_SECONDS
    return TaskResponseModel.model_validate(
        await workflow.execute_child_workflow(
            workflow=constants.WORKFLOW_EXECUTE_AGENT,
            args=[agent_config],
            task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
            id=f"{name}-{workflow.info().workflow_id}" if name else None,
        ),
    )
